recipes/jemalloc/all/conanfile.py

```diff
@@ -81,20 +84,21 @@ def validate(self):
             raise ConanInvalidConfiguration("Only Release and Debug build_types are supported")
         if self.settings.compiler == "Visual Studio" and self.settings.arch not in ("x86_64", "x86"):
             raise ConanInvalidConfiguration("Unsupported arch")
-        if self.settings.compiler == "clang" and tools.Version(self.settings.compiler.version) <= "3.9":
+        if self.settings.compiler == "clang" and Version(self.settings.compiler.version) <= "3.9":
             raise ConanInvalidConfiguration("Unsupported compiler version")
+        if self.settings.os == "Macos" and self.settings.arch not in ("x86_64", "x86"):
+            raise ConanInvalidConfiguration("Unsupported arch")
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+        if self._settings_build.os == "Windows" and not tools_legacy.get_env("CONAN_BASH_PATH"):
```

Comment:
```suggestion
        if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```
msys2 package has been updated to used only Conan conf: https://github.com/conan-io/conan-center-index/blob/master/recipes/msys2/all/conanfile.py#L194